engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/kor/KorRRNValidator.java

```diff
@@ -0,0 +1,150 @@
+
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.constraintvalidators.hv.kor;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+import org.hibernate.validator.constraints.kor.KorRRN;
+import org.hibernate.validator.internal.util.Contracts;
+import org.hibernate.validator.internal.util.ModUtil;
+
+/**
+ * Checks that a given character sequence is a valid RRN.
+ *
+ * @author Taewoo Kim
+ * @see <a href="https://www.law.go.kr/LSW/lsInfoP.do?lsId=008230&ancYnChk=0#0000">Korean Resident Registration Act Implementation Rules</a>
+ */
+public class KorRRNValidator implements ConstraintValidator<KorRRN, CharSequence> {
+
+	private static final Function<Supplier<String>, String> REPLACE_HYPEN =
+			rrn -> rrn.get().replace( "-", "" );
+	private static final List<Integer> GENDER_DIGIT = List.of( 1, 2, 3, 4 );
+	// Check sum weight for ModUtil
+	private static final int[] CHECK_SUM_WEIGHT = new int[] { 5, 4, 3, 2, 9, 8, 7, 6, 5, 4, 3, 2 };
+	// index of the digit representing the gender
+	private static final int GENDER_DIGIT_INDEX = 6;
+
+	private RRNValidationAlgorithm rrnValidationAlgorithm;
+
+	@Override
+	public void initialize(KorRRN constraintAnnotation) {
+		this.rrnValidationAlgorithm = RRNValidationAlgorithm.from( constraintAnnotation.validateCheckDigit() );
+	}
+
+	@Override
+	public boolean isValid(CharSequence rrnValue, ConstraintValidatorContext context) {
+		if ( rrnValue == null ) {
+			return true;
+		}
+		return rrnValidationAlgorithm.isValid( REPLACE_HYPEN.apply( rrnValue::toString ) );
```

Comment:
We can do a simple tostring-replace, to avoid unnecessary lambda functions.
```suggestion
		return rrnValidationAlgorithm.isValid( rrnValue.toString().replace( "-", "" ) );
```